akka-actor-tests/src/test/scala/akka/actor/ExtensionSpec.scala

```diff
@@ -20,15 +20,15 @@ import akka.testkit.TestKit._
 class JavaExtensionSpec extends JavaExtension with JUnitSuiteLike
 
 object TestExtension extends ExtensionId[TestExtension] with ExtensionIdProvider {
-  def lookup = this
+  def lookup() = this
   def createExtension(s: ExtendedActorSystem) = new TestExtension(s)
 }
 
 // Dont't place inside ActorSystemSpec object, since it will not be garbage collected and reference to system remains
 class TestExtension(val system: ExtendedActorSystem) extends Extension
 
 object FailingTestExtension extends ExtensionId[FailingTestExtension] with ExtensionIdProvider {
-  def lookup = this
+  def lookup() = this
```

Comment:
In #28991 we decided to remove `()` from the parent, rather than adding them where missing. The reasoning is that this method should return a constant, so (despite the name) it is more like a field than a method.

Reply:
Removing `()` from `ExtensionIdProvider.lookup()` is not source compatible.
The following code will not compile after that change:
```scala
def x: ExtensionIdProvider = ???
x.lookup()
```

Reply:
'lookup' isn't meant to be called from user code, though.

Nonetheless, I guess I can live with adding the `()` rather than removing them ;)